tests/test_date.py

```diff
@@ -118,163 +192,230 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
 
-        twelfth = datetime(2014, 2, 12)
-        datetime_mock = Mock(wraps=datetime)
-        datetime_mock.utcnow = Mock(return_value=twelfth)
+    def when_date_is_parsed_with_format(self, some_date, some_format):
+        self.dt_data = date.parse_with_formats(some_date, [some_format])
 
-        with patch('dateparser.date_parser.datetime', new=datetime_mock):
-            dt_data = date.parse_with_formats('February 2014', ['%B %Y'])
+    def then_should_use_current_year_for_dates_without_year(self):
+        self.assertEquals(datetime(self.year, self.month, self.day).date(), self.dt_data['date_obj'].date())
 
-        self.assertIsNotNone(dt_data)
-        self.assertEquals('month', dt_data['period'])
-        self.assertEquals(datetime(2014, 2, 28).date(), dt_data['date_obj'].date())
+    def then_parsed_date_is(self):
+        self.assertEquals(self.date_obj, self.dt_data['date_obj'].date())
 
+    def then_shouldnt_parse_invalid_date(self):
+        self.parsed_invalid_date = date.parse_with_formats(self.invalid_date,
+                                                           ['%Y-%m-%d'])['date_obj']
+        self.assertIsNone(self.parsed_invalid_date)
 
-class DateDataParserTest(unittest.TestCase):
+    def then_dates_is_not_none(self):
+        self.assertIsNotNone(self.dt_data)
 
+    def then_period_equal(self, month):
+        self.assertEquals(month, self.dt_data['period'])
+
+    def then_dates_are_equal(self):
+        self.assertEquals(self.date_obj, self.dt_data['date_obj'].date())
+
+
+class DateDataParserTest(BaseTestCase):
     def setUp(self):
+        super(DateDataParserTest, self).setUp()
         self.parser = date.DateDataParser()
+        self.date_format = NotImplemented
 
     def check_equal(self, first, second, date_string):
         self.assertEqual(first, second, "%s != %s for date_string:  '%s'" %
                          (repr(first), repr(second), date_string))
 
     def test_time_in_today_should_return_today(self):
-        date_string = '10:04am EDT'
-        today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
-
-        date_data = self.parser.get_date_data(date_string)
-        self.assertIsNotNone(date_data['date_obj'])
-        self.assertEqual(today.date(), date_data['date_obj'].date())
+        self.given_date_string('10:04am EDT')
+        self.given_today()
+        self.when_date_data_is_parsed()
+        self.then_date_was_parsed()
+        self.then_date_was_parsed()
 
     @parameterized.expand([
-        param('today'),
-        param('Today'),
-        param('TODAY'),
-        param('Сегодня'),
-        param('Hoje'),
-        param('Oggi'),
+        param('today', days_ago=0),
+        param('Today', days_ago=0),
+        param('TODAY', days_ago=0),
+        param('Сегодня', days_ago=0),
+        param('Hoje', days_ago=0),
+        param('Oggi', days_ago=0),
+        param('yesterday', days_ago=1),
```

Comment:
Let's add separator comments `# Today / Yesterday / Day before yesterday` like [here](https://github.com/scrapinghub/dateparser/blob/82db82049daf9fb62216182c2198d8a44c84df7c/tests/test_freshness_date_parser.py#L93) to make it more readable.